Automattic/wp-calypso
# Billing: Allow adding new cards from rebuilt update payment method screen

#### Changes proposed in this Pull Request

* In #48187, we added some UI elements to a new version of the Update Payment Method screen aimed at allowing users to change the payment method on a subscription to PayPal or a stored card, in addition to a new card; however that PR did not include any of the functionality. This companion PR adds in that functionality for new cards.

#### Testing instructions

* On a site with at least one existing purchase, navigate to the billing management page at `Plans > Billing` and select an upgrade to manage.
* Click through to the "Change Payment Method" screen.
* Verify that the currently assigned stored card is preselected.
* Select the "Credit or debit card" option and complete the form with valid (test) card details. Click "Save card".
* Check that you are redirected to the billing page and see a calypso success notification, and that the billing page reflects your newly assigned payment method.
* Navigate back to the "Update Payment Method" screen and verify that your newly added card (1) appears in the list and (2) is preselected.
* Repeat with some special case cards: one requiring 3DS authentication (success and failure) and one returning an error code from the payment gateway. In case of error, verify that the screen becomes interactable again.

This also touches checkout components, so we should check for regressions there. It should be enough to make sure you can complete a purchase.
